src/test/java/com/bbaga/githubscheduledreminderapp/domain/sources/github/AuthorFilterTest.java

```diff
@@ -0,0 +1,142 @@
+        Mockito.when(mockExcludeUser.getLogin()).thenReturn(EXCLUDE_USERNAME);
+        Mockito.when(mockUnkownUser.getLogin()).thenReturn(UNKNOWN_USERNAME);
+    }
+
+    @Test
+    void filter_include() throws IOException {
+        AuthorFilterConfig config = new AuthorFilterConfig();
+        config.setIncludeAuthors(List.of(INCLUDE_USERNAME));
+        AuthorFilter filter = new AuthorFilter();
+        filter.configure(config);
+
+        GitHubPullRequest pr = mock(GitHubPullRequest.class);
+        Mockito.when(pr.isDraft()).thenReturn(false);
+        Mockito.when(pr.getUpdatedAt()).thenReturn(new Date());
+        Mockito.when(pr.getUser()).thenReturn(mockIncludeUser);
+        Assertions.assertFalse(filter.filter(pr));
+
+        config.setIncludeAuthors(List.of());
+        Assertions.assertFalse(filter.filter(pr));
+    }
+
+    @Test
+    void filter_exclude() throws IOException {
+        AuthorFilterConfig config = new AuthorFilterConfig();
+        config.setExcludeAuthors(List.of(EXCLUDE_USERNAME));
+        AuthorFilter filter = new AuthorFilter();
+        filter.configure(config);
+
+        GitHubPullRequest pr = mock(GitHubPullRequest.class);
+        Mockito.when(pr.isDraft()).thenReturn(false);
+        Mockito.when(pr.getUpdatedAt()).thenReturn(new Date());
+        Mockito.when(pr.getUser()).thenReturn(mockExcludeUser);
+        Assertions.assertTrue(filter.filter(pr));
+
+        config.setExcludeAuthors(List.of());
+        Assertions.assertFalse(filter.filter(pr));
+    }
+
+    @Test
+    void filter_include_exclude() throws IOException {
+        AuthorFilterConfig config = new AuthorFilterConfig();
+        config.setIncludeAuthors(List.of(INCLUDE_USERNAME));
+        config.setExcludeAuthors(List.of(EXCLUDE_USERNAME));
+        AuthorFilter filter = new AuthorFilter();
+        filter.configure(config);
+
+        GitHubPullRequest pr = mock(GitHubPullRequest.class);
+        Mockito.when(pr.isDraft()).thenReturn(true);
+        Mockito.when(pr.getUpdatedAt()).thenReturn(new Date());
+        Mockito.when(pr.getUser()).thenReturn(mockIncludeUser);
+        Assertions.assertFalse(filter.filter(pr));
+
+        Mockito.when(pr.getUser()).thenReturn(mockExcludeUser);
+        Assertions.assertTrue(filter.filter(pr));
+
+        Mockito.when(pr.getUser()).thenReturn(mockUnkownUser);
+        Assertions.assertTrue(filter.filter(pr));
+    }
+
+    @Test
+    void filter_include_exclude_same_user() throws IOException {
+        AuthorFilterConfig config = new AuthorFilterConfig();
+        config.setIncludeAuthors(List.of(INCLUDE_USERNAME));
+        config.setExcludeAuthors(List.of(INCLUDE_USERNAME));
+        AuthorFilter filter = new AuthorFilter();
+        filter.configure(config);
+
+        GitHubPullRequest pr = mock(GitHubPullRequest.class);
+        Mockito.when(pr.isDraft()).thenReturn(true);
+        Mockito.when(pr.getUpdatedAt()).thenReturn(new Date());
```

Comment:
The draft flag and the updatedAt TS doesn't seem to be relevant for these tests.

Reply:
Hmm normally mockito yells about over mocking so i figured they were needed.... Removed.